Check update stagger

artushin opened the pull request:
Adding an optional stagger to the CheckUpdateInterval so that when consul runs in an environment with very frequency checks, all of those checks are not flushed at the same time when the CheckUpdateInterval expires. The stagger has a configurable on/off as "check_update_stagger" in the config and staggers between .5 and 1.5 of the CheckUpdateInterval.

ryanbreen commented:
You might want to look at the existing randomStagger: https://github.com/hashicorp/consul/blob/master/command/agent/util.go#L40

ryanbreen commented:
Also, not sure how @ryanuber and @armon feel about this, but I'm not convinced this needs to be configurable.  A little bit of stagger by default seems like the right approach.

artushin commented:
Hah, yeah, I actually pulled the logic from that function but for some reason though it was in a different package. Updated that.

@ryanuber and @armon, let me know what you think about it being configurable. I added it so that the behavior doesn't change for existing users unexpectedly but it may be a small change that would actually help everyone.

ryanuber commented:
So the situation this would help under would be a server with a very large amount of checks, all with frequently changing output and stable statuses. Since this is such a minor optimization, I don't think we need to introduce a configuration option to enable/disable it.

artushin commented:
Dropped the stagger on/off config.

armon commented:
Thanks!
